cpp/log/leveldb_db-inl.h

```diff
@@ -47,18 +47,37 @@ std::unique_ptr<const leveldb::FilterPolicy> BuildFilterPolicy() {
 #endif
 
 
-std::string FormatSequenceNumber(const int64_t seq) {
-  return std::to_string(seq);
+std::string IndexToKey(int64_t index) {
+  CHECK_EQ(8, sizeof(index));
+  const char nibble[] = "0123456789abcdef";
+  char index_str[sizeof(index) * 2];
```

Comment:
I'm going right to left, and `std::string` doesn't let you easily overwrite its data... My first cut had an `std::string`, and did `index_str = index_str + nibble[index & 0xf]`, which has an implicit temporary, and a copy, and... ugh. :stuck_out_tongue: 

I could do something fancier here, stand by.